lib/raty.js

```diff
@@ -0,0 +1,784 @@
+    });
+  }
+
+  // TODO: model spec
+  _bindOver() {
+    var that = this;
+    var action = that.opt.half ? 'mousemove' : 'mouseover';
+
+    this.stars.forEach((value) => {
+      value.addEventListener(action, function (evt) {
+        var score = that._getScoreByPosition(evt, value);
+        that._fill(score);
+
+        if (that.opt.half) {
+          that._roundStars(score, evt);
+          that._setTitle(score, evt);
+          that.self.dataset.score = score;
+        }
+
+        that._target(score, evt);
+
+        if (that.opt.mouseover) {
+          that.opt.mouseover.call(that.self, score, evt);
+        }
+      });
+    });
+  }
+
+  // TODO: model spec
+  _bindOverCancel() {
+    var that = this;
+
+    this.cancelButton.addEventListener('mouseover', function (evt) {
+      if (that.self.dataset.readOnly && !that.self.dataset.readOnly.toBoolean()) {
+        return;
+      }
+
+      var starOff = that.opt.path + that.opt.starOff;
+      var icon = that.opt.cancelOn;
+
+      if (that.opt.starType === 'img') {
+        that.stars.forEach((value) => {
+          value.src = starOff;
+        });
+      } else {
+        icon = that.opt.cancelClass + ' ' + icon;
+        that.stars.forEach((value) => {
+          value.className = starOff;
+        });
+      }
+
+      that._setIcon(that.cancelButton, icon);
+      that._target(null, evt);
+
+      if (that.opt.mouseover) {
+        that.opt.mouseover.call(that.self, null);
+      }
+    });
+  }
+
+  // TODO: model spec
+  _buildScoreField() {
+    const input = document.createElement('input');
+    input.name = this.opt.scoreName;
+    input.type = 'hidden';
+    this.self.appendChild(input);
+    return input;
+  }
+
+  // TODO: model spec
+  _createCancel() {
+    let icon = this.opt.path + this.opt.cancelOff;
+    let button = document.createElement(this.opt.starType);
+    const attributes = {
+      title: this.opt.cancelHint,
+      class: this.opt.cancelClass,
+    };
+
+    if (this.opt.starType === 'img') {
+      attributes.src = icon;
+      attributes.alt = 'x';
+    } else {
+      attributes['data-alt'] = 'x';
+      attributes.class = icon;
+    }
+    for (const key in attributes) {
+      button.setAttribute(key, attributes[key]);
+    }
+    if (this.opt.cancelPlace === 'left') {
+      this.self.prepend('\u00A0');
+      this.self.prepend(button);
+    } else {
+      this.self.append('\u00A0');
+      this.self.appendChild(button);
+    }
+    button.id = 'raty-cancel-btn';
+    this.cancelButton = document.querySelector('#raty-cancel-btn');
+  }
+
+  // TODO: model spec
+  _createScore() {
+    var score = document.querySelector(this.opt.targetScore);
+
+    this.scoreField = score ?? this._buildScoreField();
+  }
+
+  _createStars() {
+    for (var i = 1; i <= this.opt.number; i++) {
+      var attributes = this._attributesForIndex(i);
+      const img = document.createElement("img");
+      const element = document.createElement(this.opt.starType);
```

Comment:
Can we just use this and remove the above? The `starType` will already create the correct one without conditional.